iroh-net/src/derp/server.rs

```diff
@@ -500,8 +501,18 @@ where
                        }
                        ServerMessage::CreateClient(client_builder) => {
                            inc!(Metrics, accepts);
+
                            tracing::trace!("create client: {:?}", client_builder.key);
                            let key = client_builder.key;
+
+                           report_usage_stats(&UsageStatsReport::new(
+                                "derp_accepts".to_string(),
+                                self.key.to_string(),
+                                1,
+                                None, // TODO(arqu): attribute to user id; possibly with the re-introduction of request tokens or other auth
+                                Some(key.to_string()),
+                            )).await.unwrap_or_else(|e| tracing::warn!("error reporting usage stats: {:?}", e));
```

Comment:
For metrics-like APIs it might be better to make the functions infallible and handle the error reporting in the function itself.  I assume this will end up being strewn around the code in many places?